src/tests/Microsoft.Diagnostics.Monitoring.EventPipe/GlobMatcherTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.AspNet;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.UnitTests
+{
+    public class GlobMatcherTests
```

Comment:
Could we add some negative tests like trying to specify `***` as a pattern or other illegal patterns? The `RegularExpressions.Regex` constructor can throw if you provide invalid regular expressions.

Reply:
The code will actually escape all regex first and then only substitute regex for recognized patterns, so it shouldn't be possible to provide bad expressions to the ctor.

Having said that, expressions like *** and ***/ will cause redundant regex expressions and are difficult to understand so it might be good to validate them out anyway.